tests/test_readwrite.py

```diff
@@ -40,9 +47,29 @@ def factory_test_graph(ir: nir.NIRGraph):
     assert_equivalence(ir, ir2)
 
 
+def factory_test_metadata(node):
+    def compare_dicts(d1, d2):
+        for k, v in d1.items():
+            if isinstance(v, np.ndarray):
+                assert np.array_equal(v, d2[k])
+            elif isinstance(v, bytes):
+                assert v.decode("utf8") == d2[k]
+            else:
+                assert v == d2[k]
+
+    metadata = {"some": "metadata", "with": 2, "data": np.array([1, 2, 3])}
+    node.metadata = metadata
+    compare_dicts(node.metadata, metadata)
+    tmp = tempfile.mktemp()
```

Comment:
tempfile.mktemp is actually marked deprecated. They recommend tempfile.mkstemp as more secure variant:
https://docs.python.org/3/library/tempfile.html#tempfile.mktemp

Furthermore, neither mktemp nor mkstemp automatically remove the file. 
I would instead recommend tempfile.TemporaryFile which can be used as context manager, see example here:
https://docs.python.org/3/library/tempfile.html#tempfile-examples
```Python
# create a temporary file using a context manager
with tempfile.TemporaryFile() as fp:
    fp.write(b'Hello world!')
    fp.seek(0)
    fp.read()
# file is now closed and removed
```

Reply:
Thank you for the input. And, it appears that [h5py also supports temporary files/RawIOBase files as inputs](https://docs.h5py.org/en/stable/high/file.html#python-file-like-objects). I updated the test